synapse/federation/federation_server.py

```diff
@@ -652,19 +652,27 @@ async def on_send_join_request(
         )
 
         prev_state_ids = await context.get_prev_state_ids()
-        state_ids = list(prev_state_ids.values())
-        auth_chain = await self.store.get_auth_chain(room_id, state_ids)
-        state = await self.store.get_events(state_ids)
+        state_event_ids = prev_state_ids.values()
 
+        auth_chain_event_ids = await self.store.get_auth_chain_ids(
+            room_id, state_event_ids
+        )
+
+        auth_chain_events = await self.store.get_events_as_list(auth_chain_event_ids)
+        state_events = await self.store.get_events_as_list(state_event_ids)
```

Comment:
`get_events_as_list` is equivalent to `get_events`, except that it doesn't build a dict of the results. Since we only ever used the `.values` of the result, the dict-building was entirely redundant. (The only thing it could plausibly have done was dedup the results, but since `state_event_ids` is a set anyway, there couldn't have been any dups in the first place)